src/main/java/org/jenkinsci/plugins/saml/BundleKeyStore.java

```diff
@@ -0,0 +1,277 @@
+package org.jenkinsci.plugins.saml;
+
+import sun.security.x509.AlgorithmId;
+import sun.security.x509.CertificateAlgorithmId;
+import sun.security.x509.CertificateSerialNumber;
+import sun.security.x509.CertificateValidity;
+import sun.security.x509.CertificateVersion;
+import sun.security.x509.CertificateX509Key;
+import sun.security.x509.X500Name;
+import sun.security.x509.X509CertImpl;
+import sun.security.x509.X509CertInfo;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.InvalidKeyException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.SecureRandom;
+import java.security.SignatureException;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Base64;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Random;
+import java.util.logging.Logger;
+
+/**
+ * Pac4j requires to set a keystore for encryption operations,
+ * the plugin generate an automatic keystore or it it is not possible uses a keystore bundle on the plugin.
+ * The generated key is valid for a day, when expires it is generated a new one on the same keystore.
+ * A new key store is created when you restart Jenkins or if is not possible to access to the created.
+ * @see <a href="http://www.pac4j.org/1.9.x/docs/clients/saml.html">pac4j - Authentication mechanisms: SAML</a>
+ */
+public class BundleKeyStore {
+    public static final String PAC4J_DEMO_PASSWD = "pac4j-demo-passwd";
+    public static final String PAC4J_DEMO_KEYSTORE = "resource:samlKeystore.jks";
+    public static final String SIG_ALG = "SHA1WithRSA";
+    public static final String KEY_ALG = "RSA";
+    public static final Long KEY_VALIDITY = 1 * 24L * 60L * 60L;
+
+    private static final Logger LOG = Logger.getLogger(BundleKeyStore.class.getName());
+
+    private String keystorePath = PAC4J_DEMO_KEYSTORE;
+    private String ksPassword = PAC4J_DEMO_PASSWD;
+    private String ksPkPassword = PAC4J_DEMO_PASSWD;
+    private Date dateValidity;
+    private File keystore;
+
+    public BundleKeyStore() {
+    }
+
+    /**
+     * initialized the keystore, it tries to create a keystore in a file,
+     * if it fails load the settings of the demo keystore.
+     */
+    public synchronized void init() {
+        try {
+
+            if(keystore == null){
+                keystore = File.createTempFile("saml-jenkins-keystore-", ".jks");
```

Comment:
it is a temp file, it the OS it is proper configured /tmp has the flag "t" , so only the user that creates it should read it, additionally, the password for the key store is a 256bytes password